spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -0,0 +1,158 @@
+			+ "EXIT_MESSAGE = ?, LAST_UPDATED = ?, STATUS_CODE = ? "
+			+ "where TASK_EXECUTION_ID = ?";
+
+
+	private static final String DEFAULT_TABLE_PREFIX = "";
+
+	private String tablePrefix = DEFAULT_TABLE_PREFIX;
+
+	private JdbcOperations jdbcTemplate;
+
+	public JdbcTaskExecutionDao(DataSource dataSource) {
+		Assert.notNull(dataSource);
+		this.jdbcTemplate = new JdbcTemplate(dataSource);
+	}
+
+	@Override
+	public void saveTaskExecution(TaskExecution taskExecution) {
+		validateTaskExecution(taskExecution);
+
+		Object[] parameters = new Object[]{ taskExecution.getExecutionId(),
+				taskExecution.getStartTime(), taskExecution.getEndTime(),
+				taskExecution.getTaskName(), taskExecution.getExitCode(),
+				taskExecution.getExitMessage(), new Date(),
+				taskExecution.getStatusCode() };
+		int addCount = jdbcTemplate.update(
+				getQuery(SAVE_TASK_EXECUTION),
+				parameters,
+				new int[]{ Types.VARCHAR, Types.TIMESTAMP, Types.TIMESTAMP, Types.VARCHAR,
+						Types.INTEGER, Types.VARCHAR, Types.TIMESTAMP, Types.VARCHAR });
+		insertJobParameters(taskExecution.getExecutionId(), taskExecution.getParameters());
+	}
+
+	@Override
+	public void updateTaskExecution(TaskExecution taskExecution) {
+
+		validateTaskExecution(taskExecution);
+
+		// Check if given TaskExecution's Id already exists, if none is found
+		// it is invalid and an exception should be thrown.
+		if (jdbcTemplate.queryForObject(getQuery(CHECK_TASK_EXECUTION_EXISTS), Integer.class,
+				new Object[]{ taskExecution.getExecutionId() }) != 1) {
+			throw new IllegalStateException("Invalid TaskExecution, ID " + taskExecution.getExecutionId() + " not found.");
+		}
+
+		Object[] parameters = new Object[]{ taskExecution.getStartTime(), taskExecution.getEndTime(),
+				taskExecution.getTaskName(), taskExecution.getExitCode(),
+				taskExecution.getExitMessage(), new Date(), taskExecution.getStatusCode(),
+				taskExecution.getExecutionId() };
+		int count = jdbcTemplate.update(
+				getQuery(UPDATE_TASK_EXECUTION),
+				parameters,
+				new int[]{ Types.TIMESTAMP, Types.TIMESTAMP, Types.VARCHAR, Types.INTEGER,
+						Types.VARCHAR, Types.TIMESTAMP, Types.VARCHAR, Types.VARCHAR });
+	}
+
+	/**
+	 * Public setter for the table prefix property. This will be prefixed to all
+	 * the table names before queries are executed. Defaults to
+	 * {@link #DEFAULT_TABLE_PREFIX}.
+	 *
+	 * @param tablePrefix the tablePrefix to set
+	 */
+	public void setTablePrefix(String tablePrefix) {
+		this.tablePrefix = tablePrefix;
+	}
+
+	/**
+	 * Validate TaskExecution. At a minimum a startTime.
+	 *
+	 * @param taskExecution the taskExecution to be evaluagted.
+	 */
+	private void validateTaskExecution(TaskExecution taskExecution) {
+		Assert.notNull(taskExecution, "taskExecution should not be null");
+		Assert.notNull(taskExecution.getExecutionId(), "taskExecutionId should not be null");
+		Assert.notNull(taskExecution.getStartTime(), "TaskExecution start time cannot be null.");
+	}
```

Comment:
Shouldn't this logic live in the `SimpleTaskRepository` so it's consistent across DAO uses?